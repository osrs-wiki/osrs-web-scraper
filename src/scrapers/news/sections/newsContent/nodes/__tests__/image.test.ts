import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import fs from "fs";
import parse from "node-html-parser";

import imageParser from "../image";

describe("image node", () => {
  test("Image node should parse and render", () => {
    const existsSyncSpy = jest.spyOn(fs, "existsSync");
    const mkdirSyncSpy = jest.spyOn(fs, "mkdirSync");

    const root = parse(
      '<image href="https://test.com/image.png" width="200" />'
    );

    const builder = new MediaWikiBuilder();
    builder.addContents(
      [imageParser(root.firstChild, { title: "test-title" })].flat()
    );
    expect(builder.build()).toMatchSnapshot();

    expect(existsSyncSpy).toHaveBeenCalledWith(`./out/news/test-title`);
    expect(mkdirSyncSpy).toHaveBeenCalledWith(`./out/news/test-title`, {
      recursive: true,
    });
  });
});
