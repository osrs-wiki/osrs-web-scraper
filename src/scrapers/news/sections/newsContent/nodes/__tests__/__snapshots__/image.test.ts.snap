// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image node Image node should parse and render 1`] = `
"[[File:test-title (1).png|200px]]
"
`;
